source/visionEnhancementProviders/NVDAHighlighter.py

```diff
@@ -316,50 +316,57 @@ def onPanelActivated(self):
 		self.lastControl = self.optionsText
 
 	def _updateEnabledState(self):
-		settings = self._getSettingsStorage()
+		settingsStorage = self._getSettingsStorage()
 		settingsToTriggerActivation = [
-			settings.highlightBrowseMode,
-			settings.highlightFocus,
-			settings.highlightNavigator,
+			settingsStorage.highlightBrowseMode,
+			settingsStorage.highlightFocus,
+			settingsStorage.highlightNavigator,
 		]
-		if any(settingsToTriggerActivation):
-			if all(settingsToTriggerActivation):
-				self._enabledCheckbox.Set3StateValue(wx.CHK_CHECKED)
-			else:
-				self._enabledCheckbox.Set3StateValue(wx.CHK_UNDETERMINED)
-			self._ensureEnableState(True)
+		isAnyEnabled = any(settingsToTriggerActivation)
+		if all(settingsToTriggerActivation):
+			self._enabledCheckbox.Set3StateValue(wx.CHK_CHECKED)
+		elif isAnyEnabled:
+			self._enabledCheckbox.Set3StateValue(wx.CHK_UNDETERMINED)
 		else:
 			self._enabledCheckbox.Set3StateValue(wx.CHK_UNCHECKED)
-			self._ensureEnableState(False)
 
-	def _ensureEnableState(self, shouldBeEnabled: bool):
+		if not self._ensureEnableState(isAnyEnabled) and isAnyEnabled:
+			self._onEnableFailure()
+
+	def _onEnableFailure(self):
+		""" Initialization of Highlighter failed. Reset settings / GUI
+		"""
+		settingsStorage = self._getSettingsStorage()
+		settingsStorage.highlightBrowseMode = False
+		settingsStorage.highlightFocus = False
+		settingsStorage.highlightNavigator = False
+		self.updateDriverSettings()
+		self._updateEnabledState()
+
+	def _ensureEnableState(self, shouldBeEnabled: bool) -> bool:
 		currentlyEnabled = bool(self._providerControl.getProviderInstance())
 		if shouldBeEnabled and not currentlyEnabled:
-			self._providerControl.startProvider()
+			return self._providerControl.startProvider()
 		elif not shouldBeEnabled and currentlyEnabled:
-			self._providerControl.terminateProvider()
+			return self._providerControl.terminateProvider()
+		return True

```

Comment:
The main issue: `_ensureEnableState` was ignoring the return value of `startProvider()`. When starting the provider failed it was not unchecking "enable". To help ensure this, I extracted the logic for a failed init (reset config, call methods to reset GUI) and used it where we must handle failures.